source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -165,3 +165,330 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
     m_hAlignment = value;
 }
 
+bool TextBlock::scanForISO8601(std::string::const_iterator &itr, std::string::const_iterator &end, 
```

Comment:
Could we use the stl regex library to avoid doing the parsing ourselves?

Reply:
i think we can;  I thought it would be simple enough do parsing ourselves; after start working on it; i didn't get chance to think back on regex until finished.

---
In reply to: [148398764](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148398764) [](ancestors = 148398764)

Reply:
Let's look into changing this to regex in order to simplify maintainability going forward.

---
In reply to: [148403023](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148403023) [](ancestors = 148403023,148398764)

Reply:
i think that is right approach for maintenance, can we do it after November since we have time constraint? 

---
In reply to: [148415965](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148415965) [](ancestors = 148415965,148403023,148398764)

Reply:
I think if we don't do it now we won't do it. I believe using regex would simplify this code a lot and is worth doing now.

---
In reply to: [148613565](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148613565) [](ancestors = 148613565,148415965,148403023,148398764)

Reply:
let me work on it

---
In reply to: [148619282](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148619282) [](ancestors = 148619282,148613565,148415965,148403023,148398764)

Reply:
done

---
In reply to: [148639830](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r148639830) [](ancestors = 148639830,148619282,148613565,148415965,148403023,148398764)